### PR TITLE
Add IssueMarker decoration module

### PR DESCRIPTION
**Pull request by ndoschek**
- Introduce glspDecorationModule
- Use Theia issue marker icons as GlspIssueMarkers
- Add transparent svg rectangle to improve IssueMarker mouseover experience
- Fix applying severity CSS classes to model elements
   - Use maximum severity as CSS class (similar to IssueMarkers)
- Extend IssueMarker tooltip message for better readability

Please also see my PR in glsp-examples, here is a short preview with different IssueMarkers:

glspdecorationmodule_example

### CODE REVIEW

File: src/features/decoration/decoration-placer.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+@injectable()
+export class GlspDecorationPlacer extends DecorationPlacer {
+
+    protected getPosition(element: SModelElement & Decoration): Point {
+        if (element instanceof SChildElement && element.parent instanceof SRoutableElement) {
+            return super.getPosition(element);
+        }
+        if (isSizeable(element))
+            return {
+                x: 12 * element.bounds.width,
```
Comment (tortmayr) [inline src/features/decoration/decoration-placer.ts:37]: Where are these "magic numbers" coming from (12 rsp. 10). We should at least refactor them into static constants

File: src/features/validation/validate.ts [hunk 2]
```diff
@@ -208,8 +208,24 @@ export class ApplyMarkersCommand extends FeedbackCommand {
     }
 }
 
-function addCSSClassToIssueParent(modelElement: SParentElement, issueMarker: SIssueMarker) {
-    addCssClasses(modelElement, [getSeverity(issueMarker)]);
+function addMaxSeverityCSSClassToIssueParent(modelElement: SParentElement, issueMarker: SIssueMarker) {
+    const severityCSSClass = getSeverity(issueMarker);
+
+    if (modelElement.cssClasses === undefined) {
```
Comment (tortmayr) [inline src/features/validation/validate.ts:214]: Your can ommit the  `=== undefined` here by using `if (!modelElement.cssClasses)`

File: src/features/validation/validate.ts [hunk 3]
```diff
@@ -208,8 +208,24 @@ export class ApplyMarkersCommand extends FeedbackCommand {
...
+        modelElement.cssClasses = [severityCSSClass];
+    } else {
+        modelElement.cssClasses.push(severityCSSClass);
+        modelElement.cssClasses = getMaxSeverity(modelElement.cssClasses);
```
Comment (tortmayr) [inline src/features/validation/validate.ts:218]: Doesn't this remove all other cssClasses that might be attached to the modelElement/issueMarker ?

Comment (ndoschek) [reply src/features/validation/validate.ts:218]: Thanks, you're right! I edited this part, now it makes sure that only severity cssClasses are replaced by the max severity cssClass of the parent modelElement.

**Pull request merged**